Leader election seems elect in endless loop

**Describe the bug**
TransactionEndToEndTest was broken/flaky (Related to #10457)
Leader election seems elect in endless loop.
I think it can work well when connecting real zk.

**To Reproduce**
run setup in TransactionEndToEndTest

**Expected behavior**
pass test

**Additional context**
```
18:14:54.813 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.813 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.814 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.814 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.814 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.817 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.817 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.817 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.819 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.820 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.821 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.821 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.821 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.822 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.823 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.823 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.823 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.824 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.824 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.825 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.825 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.825 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.826 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.827 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.827 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.828 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.828 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.828 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.829 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.830 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.830 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.830 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.831 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.831 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.831 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.832 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.833 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.833 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.834 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.834 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.834 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.835 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.835 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.836 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.837 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.837 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.837 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.837 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.838 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.838 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.839 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.839 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.840 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.842 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.842 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.842 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.843 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.843 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.843 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.844 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.844 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.845 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.847 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.847 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.820 [main] ERROR org.apache.pulsar.broker.PulsarService - org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
java.util.concurrent.CompletionException: org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292) ~[?:1.8.0_262]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308) ~[?:1.8.0_262]
	at java.util.concurrent.CompletableFuture.uniRun(CompletableFuture.java:714) ~[?:1.8.0_262]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:701) ~[?:1.8.0_262]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_262]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990) ~[?:1.8.0_262]
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$null$21(ZKMetadataStore.java:269) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_262]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_262]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.63.Final.jar:4.1.63.Final]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_262]
Caused by: org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.getException(ZKMetadataStore.java:302) ~[classes/:?]
	... 5 more
Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:118) ~[zookeeper-3.6.2.jar:3.6.2]
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:54) ~[zookeeper-3.6.2.jar:3.6.2]
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.getException(ZKMetadataStore.java:296) ~[classes/:?]
	... 5 more
18:14:54.847 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.848 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.848 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.848 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.848 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.848 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.849 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.849 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.849 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.849 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.850 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.850 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.850 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.850 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.850 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.851 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.851 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.851 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.851 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.851 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.851 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.852 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.852 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.852 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.853 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.853 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.853 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.853 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.853 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.853 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.854 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.854 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.854 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.854 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.855 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.855 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.855 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.855 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.855 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.856 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.856 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.856 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.856 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.856 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.857 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.857 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.857 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.858 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.858 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.858 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.858 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.858 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader

18:14:54.858 [metadata-store-4-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.859 [metadata-store-4-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.859 [metadata-store-44-1] INFO  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Acquired leadership on /loadbalance/leader
18:14:54.859 [metadata-store-4-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader
18:14:54.859 [metadata-store-44-1] INFO  org.apache.pulsar.broker.PulsarService - This broker was elected leader
18:14:54.859 [metadata-store-44-1] WARN  org.apache.pulsar.metadata.coordination.impl.LeaderElectionImpl - Leadership released for /loadbalance/leader


org.apache.pulsar.broker.PulsarServerException: java.util.concurrent.CompletionException: org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader

	at org.apache.pulsar.broker.PulsarService.start(PulsarService.java:806)
	at org.apache.pulsar.broker.transaction.TransactionTestBase.startBroker(TransactionTestBase.java:138)
	at org.apache.pulsar.broker.transaction.TransactionTestBase.init(TransactionTestBase.java:105)
	at org.apache.pulsar.broker.transaction.TransactionTestBase.internalSetup(TransactionTestBase.java:84)
	at org.apache.pulsar.client.impl.TransactionEndToEndTest.setup(TransactionEndToEndTest.java:94)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:61)
	at org.testng.internal.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:366)
	at org.testng.internal.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:320)
	at org.testng.internal.TestInvoker.runConfigMethods(TestInvoker.java:701)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:527)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.testng.TestRunner.privateRun(TestRunner.java:764)
	at org.testng.TestRunner.run(TestRunner.java:585)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.runSuites(TestNG.java:1069)
	at org.testng.TestNG.run(TestNG.java:1037)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:110)
Caused by: java.util.concurrent.CompletionException: org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniRun(CompletableFuture.java:714)
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:701)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$null$21(ZKMetadataStore.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.getException(ZKMetadataStore.java:302)
	... 5 more
Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /loadbalance/leader
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:118)
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:54)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.getException(ZKMetadataStore.java:296)
	... 5 more
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/namespace/OwnerShipForCurrentServerTestBase.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/namespace/OwnerShipForCurrentServerTestBase.java
@@ -40,6 +40,7 @@ import org.apache.pulsar.zookeeper.ZooKeeperClientFactory;
 import org.apache.pulsar.zookeeper.ZookeeperClientFactoryImpl;
 import org.apache.zookeeper.CreateMode;
 import org.apache.zookeeper.MockZooKeeper;
+import org.apache.zookeeper.MockZooKeeperSession;
 import org.apache.zookeeper.ZooKeeper;
 import org.apache.zookeeper.data.ACL;
 
@@ -129,8 +130,9 @@ public class OwnerShipForCurrentServerTestBase {
         // Override default providers with mocked ones
         doReturn(mockZooKeeperClientFactory).when(pulsar).getZooKeeperClientFactory();
         doReturn(mockBookKeeperClientFactory).when(pulsar).newBookKeeperClientFactory();
-        doReturn(new ZKMetadataStore(mockZooKeeper)).when(pulsar).createLocalMetadataStore();
-        doReturn(new ZKMetadataStore(mockZooKeeper)).when(pulsar).createConfigurationMetadataStore();
+        MockZooKeeperSession mockZooKeeperSession = MockZooKeeperSession.newInstance(mockZooKeeper);
+        doReturn(new ZKMetadataStore(mockZooKeeperSession)).when(pulsar).createLocalMetadataStore();
+        doReturn(new ZKMetadataStore(mockZooKeeperSession)).when(pulsar).createConfigurationMetadataStore();
         Supplier<NamespaceService> namespaceServiceSupplier = () -> spy(new NamespaceService(pulsar));
         doReturn(namespaceServiceSupplier).when(pulsar).getNamespaceServiceProvider();
 

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTestBase.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTestBase.java
@@ -53,10 +53,11 @@ import org.apache.pulsar.metadata.impl.ZKMetadataStore;
 import org.apache.pulsar.tests.TestRetrySupport;
 import org.apache.pulsar.zookeeper.ZooKeeperClientFactory;
 import org.apache.pulsar.zookeeper.ZookeeperClientFactoryImpl;
+import org.apache.zookeeper.data.ACL;
 import org.apache.zookeeper.CreateMode;
 import org.apache.zookeeper.MockZooKeeper;
+import org.apache.zookeeper.MockZooKeeperSession;
 import org.apache.zookeeper.ZooKeeper;
-import org.apache.zookeeper.data.ACL;
 
 @Slf4j
 public abstract class TransactionTestBase extends TestRetrySupport {
@@ -144,8 +145,10 @@ public abstract class TransactionTestBase extends TestRetrySupport {
         // Override default providers with mocked ones
         doReturn(mockZooKeeperClientFactory).when(pulsar).getZooKeeperClientFactory();
         doReturn(mockBookKeeperClientFactory).when(pulsar).newBookKeeperClientFactory();
-        doReturn(new ZKMetadataStore(mockZooKeeper)).when(pulsar).createLocalMetadataStore();
-        doReturn(new ZKMetadataStore(mockZooKeeper)).when(pulsar).createConfigurationMetadataStore();
+
+        MockZooKeeperSession mockZooKeeperSession = MockZooKeeperSession.newInstance(mockZooKeeper);
+        doReturn(new ZKMetadataStore(mockZooKeeperSession)).when(pulsar).createLocalMetadataStore();
+        doReturn(new ZKMetadataStore(mockZooKeeperSession)).when(pulsar).createConfigurationMetadataStore();
         Supplier<NamespaceService> namespaceServiceSupplier = () -> spy(new NamespaceService(pulsar));
         doReturn(namespaceServiceSupplier).when(pulsar).getNamespaceServiceProvider();
 

--- a/testmocks/src/main/java/org/apache/zookeeper/MockZooKeeperSession.java
+++ b/testmocks/src/main/java/org/apache/zookeeper/MockZooKeeperSession.java
@@ -219,6 +219,8 @@ public class MockZooKeeperSession extends ZooKeeper {
 
     @Override
     public String toString() {
-        return "MockZookeeperSession";
+        return "MockZooKeeperSession{" +
+                "sessionId=" + sessionId +
+                '}';
     }
 }

--- a/testmocks/src/main/java/org/apache/zookeeper/MockZooKeeperSession.java
+++ b/testmocks/src/main/java/org/apache/zookeeper/MockZooKeeperSession.java
@@ -1,0 +1,224 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.zookeeper;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiPredicate;
+import org.apache.zookeeper.AsyncCallback.Children2Callback;
+import org.apache.zookeeper.AsyncCallback.ChildrenCallback;
+import org.apache.zookeeper.AsyncCallback.DataCallback;
+import org.apache.zookeeper.AsyncCallback.StatCallback;
+import org.apache.zookeeper.AsyncCallback.VoidCallback;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Stat;
+import org.objenesis.Objenesis;
+import org.objenesis.ObjenesisStd;
+import org.objenesis.instantiator.ObjectInstantiator;
+
+/**
+ * mock zookeeper with different session based on {@link MockZooKeeper}
+ */
+public class MockZooKeeperSession extends ZooKeeper {
+
+    private MockZooKeeper mockZooKeeper;
+
+    private long sessionId = 0L;
+
+    private static final Objenesis objenesis = new ObjenesisStd();
+
+    private static final AtomicInteger sessionIdGenerator = new AtomicInteger(0);
+
+    public static MockZooKeeperSession newInstance(MockZooKeeper mockZooKeeper) {
+        ObjectInstantiator<MockZooKeeperSession> instantiator = objenesis.getInstantiatorOf(MockZooKeeperSession.class);
+        MockZooKeeperSession mockZooKeeperSession = instantiator.newInstance();
+
+        mockZooKeeperSession.mockZooKeeper = mockZooKeeper;
+        mockZooKeeperSession.sessionId = sessionIdGenerator.getAndIncrement();
+        return mockZooKeeperSession;
+    }
+
+    private MockZooKeeperSession(String quorum) throws Exception {
+        // This constructor is never called
+        super(quorum, 1, event -> {
+        });
+        assert false;
+    }
+
+    @Override
+    public int getSessionTimeout() {
+        return mockZooKeeper.getSessionTimeout();
+    }
+
+    @Override
+    public States getState() {
+        return mockZooKeeper.getState();
+    }
+
+    @Override
+    public void register(Watcher watcher) {
+        mockZooKeeper.register(watcher);
+    }
+
+    @Override
+    public String create(String path, byte[] data, List<ACL> acl, CreateMode createMode)
+            throws KeeperException, InterruptedException {
+        return mockZooKeeper.create(path, data, acl, createMode);
+    }
+
+    @Override
+    public void create(final String path, final byte[] data, final List<ACL> acl, CreateMode createMode,
+                       final AsyncCallback.StringCallback cb, final Object ctx) {
+        mockZooKeeper.create(path, data, acl, createMode, cb, ctx);
+    }
+
+    @Override
+    public byte[] getData(String path, Watcher watcher, Stat stat) throws KeeperException {
+        return mockZooKeeper.getData(path, watcher, stat);
+    }
+
+    @Override
+    public void getData(final String path, boolean watch, final DataCallback cb, final Object ctx) {
+        mockZooKeeper.getData(path, watch, cb, ctx);
+    }
+
+    @Override
+    public void getData(final String path, final Watcher watcher, final DataCallback cb, final Object ctx) {
+        mockZooKeeper.getData(path, watcher, cb, ctx);
+    }
+
+    @Override
+    public void getChildren(final String path, final Watcher watcher, final ChildrenCallback cb, final Object ctx) {
+        mockZooKeeper.getChildren(path, watcher, cb, ctx);
+    }
+
+    @Override
+    public List<String> getChildren(String path, Watcher watcher) throws KeeperException {
+        return mockZooKeeper.getChildren(path, watcher);
+    }
+
+    @Override
+    public List<String> getChildren(String path, boolean watch) throws KeeperException, InterruptedException {
+        return mockZooKeeper.getChildren(path, watch);
+    }
+
+    @Override
+    public void getChildren(final String path, boolean watcher, final Children2Callback cb, final Object ctx) {
+        mockZooKeeper.getChildren(path, watcher, cb, ctx);
+    }
+
+    @Override
+    public Stat exists(String path, boolean watch) throws KeeperException, InterruptedException {
+        return mockZooKeeper.exists(path, watch);
+    }
+
+    @Override
+    public Stat exists(String path, Watcher watcher) throws KeeperException, InterruptedException {
+        return mockZooKeeper.exists(path, watcher);
+    }
+
+    @Override
+    public void exists(String path, boolean watch, StatCallback cb, Object ctx) {
+        mockZooKeeper.exists(path, watch, cb, ctx);
+    }
+
+    @Override
+    public void exists(String path, Watcher watcher, StatCallback cb, Object ctx) {
+        mockZooKeeper.exists(path, watcher, cb, ctx);
+    }
+
+    @Override
+    public void sync(String path, VoidCallback cb, Object ctx) {
+        mockZooKeeper.sync(path, cb, ctx);
+    }
+
+    @Override
+    public Stat setData(final String path, byte[] data, int version) throws KeeperException, InterruptedException {
+        return mockZooKeeper.setData(path, data, version);
+    }
+
+    @Override
+    public void setData(final String path, final byte[] data, int version, final StatCallback cb, final Object ctx) {
+        mockZooKeeper.setData(path, data, version, cb, ctx);
+    }
+
+    @Override
+    public void delete(final String path, int version) throws InterruptedException, KeeperException {
+        mockZooKeeper.delete(path, version);
+    }
+
+    @Override
+    public void delete(final String path, int version, final VoidCallback cb, final Object ctx) {
+        mockZooKeeper.delete(path, version, cb, ctx);
+    }
+
+    @Override
+    public void multi(Iterable<org.apache.zookeeper.Op> ops, AsyncCallback.MultiCallback cb, Object ctx) {
+        mockZooKeeper.multi(ops, cb, ctx);
+    }
+
+    @Override
+    public List<OpResult> multi(Iterable<org.apache.zookeeper.Op> ops) throws InterruptedException, KeeperException {
+        return mockZooKeeper.multi(ops);
+    }
+
+    @Override
+    public void close() throws InterruptedException {
+        mockZooKeeper.close();
+    }
+
+    public void shutdown() throws InterruptedException {
+        mockZooKeeper.shutdown();
+    }
+
+    Optional<KeeperException.Code> programmedFailure(MockZooKeeper.Op op, String path) {
+        return mockZooKeeper.programmedFailure(op, path);
+    }
+
+    void maybeThrowProgrammedFailure(MockZooKeeper.Op op, String path) throws KeeperException {
+        mockZooKeeper.maybeThrowProgrammedFailure(op, path);
+    }
+
+    public void failConditional(KeeperException.Code rc, BiPredicate<MockZooKeeper.Op, String> predicate) {
+        mockZooKeeper.failConditional(rc, predicate);
+    }
+
+    public void setAlwaysFail(KeeperException.Code rc) {
+        mockZooKeeper.setAlwaysFail(rc);
+    }
+
+    public void unsetAlwaysFail() {
+        mockZooKeeper.unsetAlwaysFail();
+    }
+
+    public void setSessionId(long id) {
+        sessionId = id;
+    }
+
+    @Override
+    public long getSessionId() {
+        return sessionId;
+    }
+
+    @Override
+    public String toString() {
+        return "MockZookeeperSession";
+    }
+}

